sale_automatic_workflow/models/automatic_workflow_job.py

```diff
@@ -0,0 +1,146 @@
+        _logger.debug('Invoices to Register Payment: %s', invoices.ids)
+        for invoice in invoices:
+            partner_type = invoice.type in ('out_invoice', 'out_refund') and \
+                'customer' or 'supplier'
+            payment_type = invoice.type in ('out_invoice', 'in_refund') and \
+                'inbound' or 'outbound'
+            payment_method = 1  # TODO
+#             self.env.ref('account.account_payment_method_manual_in')
+            with commit(self.env.cr):
+                payment = self.env['account.payment'].create({
+                    'invoice_ids': [(6, 0, invoice.ids)],
+                    'amount': invoice.residual,
+                    'payment_date': fields.Date.context_today(self),
+                    'communication': invoice.reference or invoice.number,
+                    'partner_id': invoice.partner_id.id,
+                    'partner_type': partner_type,
+                    'payment_type': payment_type,
+                    'payment_method_id': payment_method,
+                })
+                payment.post()
+        return
+
+    @api.model
+    def _validate_pickings(self, picking_filter):
+        picking_obj = self.env['stock.picking']
+        pickings = picking_obj.search(picking_filter)
+        _logger.debug('Pickings to validate: %s', pickings.ids)
+        if pickings:
+            with commit(self.env.cr):
+                pickings.validate_picking()
+
+    @api.model
+    def _sale_done(self, sale_done_filter):
+        sale_obj = self.env['sale.order']
+        sales = sale_obj.search(sale_done_filter)
+        _logger.debug('Sale Orders to done: %s', sales.ids)
+        for sale in sales:
+            with commit(self.env.cr):
+                sale.action_done()
+
+    @api.model
+    def run(self):
+        """ Must be called from ir.cron """
+        sale_workflow_process = self.env['sale.workflow.process']
+        for sale_workflow in sale_workflow_process.search([]):
+            workflow_domain = [('workflow_process_id', '=', sale_workflow.id)]
+            if sale_workflow.validate_order:
+                self._validate_sale_orders(
+                    safe_eval(sale_workflow.order_filter_id.domain) +
+                    workflow_domain)
+            if sale_workflow.validate_picking:
+                self._validate_pickings(
+                    safe_eval(sale_workflow.picking_filter_id.domain) +
+                    workflow_domain)
+            if sale_workflow.create_invoice:
+                self._create_invoices(
+                    safe_eval(sale_workflow.create_invoice_filter_id.domain) +
+                    workflow_domain)
+            if sale_workflow.validate_invoice:
+                self._validate_invoices(
+                    safe_eval(
+                        sale_workflow.validate_invoice_filter_id.domain) +
+                    workflow_domain)
+            # TODO
+#             if sale_workflow.register_payment:
```

Comment:
@guewen moved the payment functionality to sale_automatic_workflow_payment_mode and created PR to your fork https://github.com/guewen/sale-workflow/pull/3